### DISCUSSION TITLE
How can I correctly use "MqttSink" in multithreading？

### DISCUSSION THREAD

**Discussion by Evan-Huang-c** [ntex-rs/ntex-mqtt#179] [Q&A]
![image](https://github.com/user-attachments/assets/19229370-8155-4ba8-a6be-335c062637ba)

We are using Rc here, not Arc. Do we need to wrap an outer layer of Arc? I think this is a bit strange

**Comment by fafhrd91** [comment]
you need two part:

1. start task in same thread as sink, read from channel in a loop and send data to sink
2. send other part of channel to different thread and use it to send data to mqtt sink.

PS ntex in general designed to be used in single threaded environment
